File: nimbus/View/MainVC.m
//
//  MainVC.m
//  nimbus
//
//  Created by Pericles Maravelakis on 19/04/2017.
//	periclesm@cloudfields.net
//	Licensed under Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)
//	https://creativecommons.org/licenses/by-sa/4.0/
//

#import "MainVC.h"
#import "DetailVC.h"

#pragma mark - Table View Controller

@implementation MainVC

#pragma mark - View Lifecycle

- (void)viewDidLoad
{
    [super viewDidLoad];
	cloudArray = [self GetCloudData];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
	if ([[segue identifier] isEqualToString:@"detailSegue"])
	{
		DetailVC *dtc = [segue destinationViewController];
		dtc.objectId = (NSString*)sender;
	}
}

#pragma mark - Data

- (NSMutableArray*)GetCloudData
{
	if (cloudArray.count > 0) [cloudArray removeAllObjects];
	
	return [[DataQuery GetSortedItemsFromEntity:@"CloudList" sortBy:@"order" ascending:YES] mutableCopy];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return cloudArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
	return [TableCellSynthesis CloudCell:tableView datasource:cloudArray index:indexPath];
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
	[tableView deselectRowAtIndexPath:indexPath animated:YES];
	CloudList *cl = cloudArray[indexPath.row];
	[self performSegueWithIdentifier:@"detailSegue" sender:cl.objectId];
}

@end

File: nimbus/View/MainVC.swift
//
//  MainVC.swift
//  nimbus
//
//  Created by Pericles Maravelakis on 1/2/20.
//  periclesm@cloudfields.net
//  Licensed under Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)
//  https://creativecommons.org/licenses/by-sa/4.0/
//

import UIKit

class MainVC: UITableViewController {
    
	let vm = MainVM()

    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "detailSegue" {
            let dvc = segue.destination as! DetailVC
			if let object = sender as? String {
				dvc.vm.cloudId = object
			}
        }
    }
    
    // MARK: - Actions
    
    @IBAction func refreshData() {
		vm.refreshData { completed in
			if completed {
				self.tableView.reloadData()
				self.refreshControl?.endRefreshing()
			}
		}
    }

    // MARK: - Table view data source

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
		return vm.clouds.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
		return TableCellSynthesis.cloudCell(for: tableView, datasource: vm.clouds, index: indexPath)
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
		let cl = vm.clouds[indexPath.row]
		self.performSegue(withIdentifier: "detailSegue", sender: cl.objectId)
    }
}
